docs/source/native/class.rst

```diff
@@ -0,0 +1,159 @@
+Classes
+=======
+
+.. note::
+
+   This is by no means complete. Please add anything you know.
+
+Declaring Classes
+-----------------
+
+To declare a class, first add the ``untyped`` keyword and the class as a global variable at file level.
+
+.. code-block:: javascript
+
+    untyped
+    global var ExampleClass
+
+The ``untyped`` declaration is required because instances have an unknown type and it's not possible to use classes as types.
+
+``global var [classname]`` represents the class. After declaring the class inside of a function you can use it anywhere. It's not possible to declare classes on local variables.
+
+You can then add the declaration inside of a function.
+
+Most classes use a constructor. A constructor is a function of the instance that gets executed on object creation.
+
+.. code-block:: javascript
+
+    void function initClient() {
```

Comment:
defining a class inside the function seems a bit odd. why?

Reply:
Only global variables, structs and functions are allowed at file level. Class declaration is only inserting the class into the variable (so not different from other logic)

For example, 
```
class Example {
    function func(){return "examplefunc";}
}
```
is the same as
```
Example <- class {
    function func(){return "examplefunc";}
}
```
[doc example](http://www.squirrel-lang.org/squirreldoc/reference/language/classes.html#class-declaration)